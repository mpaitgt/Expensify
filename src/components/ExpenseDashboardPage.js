import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseListItem from './ExpenseListItem';

const ExpenseDashboardPage = () => {
  return (
    <div>
      <ExpenseListFilters />
      <ExpenseList />
    </div>
  )
}

export default ExpenseDashboardPage;