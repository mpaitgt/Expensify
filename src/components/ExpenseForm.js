import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.expense ? props.expense.description : '',
      amount: props.expense ? (props.expense.amount / 10).toString() : '',
      note: props.expense ? props.expense.note : '',
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calendarFocused: false,
      error: ''
    };
  }

  onTextChange = (e) => {
    const { name, value } = e.target;
    this.setState(() => ({
      [name]: value
    }))
  }

  onAmountChange = (e) => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  }

  onDateChange = (createdAt) => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  }

  onFocusChange = ({focused}) => {
    this.setState(() => ({ calendarFocused: focused }));
  }

  onSubmit = (e) => {
    e.preventDefault();

    if (!this.state.description || !this.state.amount) {
      this.setState(() => ({
        error: 'Please provide a description and an amount.'
      }))
    } else {
      this.setState(() => ({
        error: ''
      }))
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };

  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input 
            name="description" 
            type="text" 
            placeholder="description" 
            autoFocus 
            value={this.state.description} 
            onChange={this.onTextChange} 
          />
          <input 
            name="amount" 
            type="text" 
            placeholder="amount" 
            value={this.state.amount} 
            onChange={this.onAmountChange} 
          />
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onFocusChange} // PropTypes.func.isRequired
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
          <textarea 
            name="note" 
            placeholder="Add a note for your expense (optional)"
            value={this.state.note} 
            onChange={this.onTextChange}  
          />
          <button>{this.props.fromPage} Expense</button>
        </form>
      </div>
    );
  };
};

export default ExpenseForm;