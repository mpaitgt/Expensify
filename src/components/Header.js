import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <nav className="header">
    <h1>Expensify</h1>
      <NavLink exact to="/" activeClassName="is-active">Home</NavLink>
      <NavLink to="/create" activeClassName="is-active">Create</NavLink>
  </nav>
)

export default Header;