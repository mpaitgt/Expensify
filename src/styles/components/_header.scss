.header {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
}

// .navigation {
//   display: flex;
//   flex-direction: row;
//   justify-content: space-between;
//   align-items: center;
//   div {
//     margin: 0 2.4rem;
//   }
// }