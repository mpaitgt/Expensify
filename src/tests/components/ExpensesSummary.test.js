import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should correctly render ExpensesSummary with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={120060} />);
  expect(wrapper).toMatchSnapshot();
});

test('Should correctly render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={5} expensesTotal={3469756} />);
  expect(wrapper).toMatchSnapshot();
});