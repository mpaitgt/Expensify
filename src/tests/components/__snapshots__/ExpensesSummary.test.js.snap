// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    5
     
    expenses
     totalling 
    $34,697.56
  </h2>
</div>
`;

exports[`Should correctly render ExpensesSummary with one expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $1,200.60
  </h2>
</div>
`;
